### ISSUE TITLE
bug: ion-img: alt text shown for brief period before src is set on inner img

### ISSUE THREAD

**Issue by HyperLife1119** [ionic-team/ionic-framework#21781]
# Feature Request

**Ionic version:**

[x] **5.x**

**Describe the Feature Request**

Currently, the `alt` attribute of `ion-img` will be displayed when the image is loading. But I think the `alt` attribute of `ion-img` should only be displayed when the image fails to load, just like the native img tag.

**Describe Preferred Solution**

**Describe Alternatives**

When loading an image, `ion-img` should show the loading state(spinner) instead of displaying the content of the `alt` attribute

**Related Code**

**Additional Context**

**Comment by liamdebeasi** [comment]
Thanks for the issue. The `alt` text currently does display when the image fails to load, just like the native img tag.

CodePen with example: https://codepen.io/liamdebeasi/pen/OJMrjbv

Can you please provide a repo with the code required to reproduce this issue?

**Comment by HyperLife1119** [comment]
I mean, the alt text is only displayed when the image fails to load. But the current situation is that the image does not fail to load, but is in the loading stage, but the alt text is still displayed.
I hope that the alt text will not be displayed when the image is loading, but only when the image fails to load.

**Comment by HyperLife1119** [comment]
I think it may be caused by ion-img not working well in animation.
Use a sliding animation to slide ion-img into the scene from outside the screen (such as routing transition animation), ion-img may not be able to detect that it has appeared on the screen, and does not actively load the image, but directly displays the alt text

**Comment by liamdebeasi** [comment]
Can you please provide a full repo I can use to reproduce this issue?

**Comment by HyperLife1119** [comment]
This project can trigger the problem that ion-img cannot actively load the image in the animation and directly display the alt text.
https://github.com/HyperLife1119/ion-img
A certain probability is required to trigger the problem, and it is easier to see the effect using a mobile phone test (must be a real mobile phone, not a virtual mobile phone)
Use the `ng serve --host=0.0.0.0` command to start the project, connect the phone to the same WiFi as the computer, and use the phone to open the web page

**Comment by HyperLife1119** [comment]
In this project, click the A/B button to switch the route, which triggers the route transition animation. The ion-img is in the animation. There is a possibility that the image cannot be loaded actively. This problem is particularly obvious on mobile devices.

**Comment by amtineo** [comment]
Same here, when loading an image the alt tag is being displayed instead of presenting it as a fallback when it couldn't be loaded. Also noticed that if the alt tag isn't set or has an empty value, the image will not load when scrolling to it (when out of the view area).

**Comment by HyperLife1119** [comment]
I also provided a code repository
https://github.com/HyperLife1119/ion-img

**Comment by liamdebeasi** [comment]
Thanks! I think I see the problem. There appears to be a brief time between when `ion-img` is added to the DOM and a callback is fired that adds the correct `src` to the inner `img` tag. In between that time, `img` does not have a valid `src`, so Chrome sees it as a broken image, and shows the alt text as a result.

**Comment by HyperLife1119** [comment]
Thanks, I thought this issue was forgotten.

**Comment by amtineo** [comment]
@liamdebeasi was this forgotten? 👎

**Comment by matEhickey** [comment]
+1 problem is still here

**Comment by Jacob-McKay** [comment]
Possibly related: https://github.com/ionic-team/ionic-framework/issues/23703

**Comment by tnoetzel** [comment]
Here's a workaround until this gets fixed:

CSS:
```
ion-img::part(image) {
    display: none;
}

ion-img.show::part(image) {
    display: block;
}
```

React TSX:
```
<IonImg onIonImgDidLoad={(e: any) => e.target.classList.add('show')} src={'yourimage.png'} alt="whatever" />
```

**Comment by colabottles** [comment]
Just to let people be aware, that is an expected behavior and has been since the HTML 1.2 spec was released. When the image is loading and it is an image that requires alt text with the alt attribute, then it should/will be seen until the image is done loading. So when internet speeds weren't fast, you could read the alt text (which was what you would see in text based browsers). 

Granted, you shouldn't see the image alt text for very long these days, but I wouldn't consider this a bug. cc: @liamdebeasi
